lib/dao.js

```diff
@@ -1659,7 +1669,7 @@ DataAccessObject.find = function find(query, options, cb) {
             });
 
             // FIXME: apply "includes" and other transforms - see allCb below
-            memory.all(modelName, ctx.query, options, cb);
+            memory.all(modelName, query, options, cb);
```

Comment:
Are you sure your new test is triggering this specific branch of code? I think the trick is to include geo query inside `where`.